lib/matplotlib/widgets.py

```diff
@@ -1487,7 +1488,10 @@ def _press(self, event):
         self.rect.set_visible(self.visible)
         if self.span_stays:
             self.stay_rect.set_visible(False)
-
+            # really force a draw so that the stay rect is not in
+            # the blit background
+            if self.useblit:
+                self.canvas.draw()
```

Comment:
Why would the background include it? The artist has been set to be
invisible, thereby making things stale. Shouldn't it then remake the
background automatically at the next draw?

On Tue, Dec 15, 2015 at 11:33 AM, Thomas A Caswell <notifications@github.com
> wrote:

> In lib/matplotlib/widgets.py
> <https://github.com/matplotlib/matplotlib/pull/5681#discussion_r47660978>:
>
> > @@ -1487,7 +1488,10 @@ def _press(self, event):
> >          self.rect.set_visible(self.visible)
> >          if self.span_stays:
> >              self.stay_rect.set_visible(False)
> > -
> > +            # really force a draw so that the stay rect is not in
> > +            # the blit background
> > +            if self.useblit:
> > +                self.canvas.draw()
>
> If the stay rectangle is present then the background that is grabbed will
> include it, by forcing the draw here it is removed from the background used
> for blitting.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/5681/files#r47660978>.
>